lib/manageiq/api/common/headers.rb

```diff
@@ -0,0 +1,44 @@
+module HeaderScope
+  thread_mattr_accessor :current
+end
+
+module ManageIQ
+  module API
+    module Common
+      module Headers
+
+        def self.current=(val)
+          validate_headers(val)
+          HeaderScope.current = val
+        end
+
+        def self.current
+          HeaderScope.current
+        end
+
+        def self.decode(headers, key)
```

Comment:
@syncrou couldn't the headers be accessed from the HeaderScope.current? why does the caller have to pass it in?

Reply:
Yes if it was set before hand.  I'm open to either: not decoding if `ManageIQ::API::Common::Headers.current` is nil or as a stand alone utility with no dependencies.  I really wanted to have the conversation around the most obvious use cases to see what everyone thought would be best.

Reply:
I agree with madhu here that it should just use current or be a private class method that can be treated in isolation.

Reply:
I wanted the validation to happen ahead of setting the accessor. Wasn't sure how else to declare the accessor and control validation ahead of setting it.

Reply:
@Fryguy do you think I shouldn't have any validation before setting `.current`? I guess I should ask that question too.